client/my-sites/earn/ads/wrapper.jsx

```diff
@@ -36,44 +34,62 @@ import {
 	getWordAdsErrorForSite,
 	getWordAdsSuccessForSite,
 } from 'calypso/state/wordads/approve/selectors';
-import { wordadsUnsafeValues } from 'calypso/state/wordads/status/schema';
 import { isSiteWordadsUnsafe } from 'calypso/state/wordads/status/selectors';
 
 import './style.scss';
 import 'calypso/my-sites/stats/stats-module/style.scss';
 
-class AdsWrapper extends Component {
-	static propTypes = {
-		adsProgramName: PropTypes.string,
-		isUnsafe: PropTypes.oneOf( wordadsUnsafeValues ),
-		requestingWordAdsApproval: PropTypes.bool.isRequired,
-		requestWordAdsApproval: PropTypes.func.isRequired,
-		section: PropTypes.string.isRequired,
-		site: PropTypes.object,
-		wordAdsError: PropTypes.string,
-		wordAdsSuccess: PropTypes.bool,
-	};
+const AdsWrapper = ( { section, children } ) => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const site = useSelector( ( state ) => getSelectedSite( state ) );
+	const siteId = useSelector( ( state ) => getSelectedSiteId( state ) );
+	const siteSlug = useSelector( ( state ) => getSelectedSiteSlug( state ) );
 
-	handleDismissWordAdsError = () => {
-		const { siteId } = this.props;
-		this.props.dismissWordAdsError( siteId );
-	};
+	const canAccessAds = useSelector( ( state ) => canAccessWordAds( state, siteId ) );
+	const wordAdsStatus = useSelector( ( state ) => getSiteWordadsStatus( state, siteId ) );
+	const hasWordAdsFeature = useSelector( ( state ) => siteHasWordAds( state, siteId ) );
+	const wordAdsError = useSelector( ( state ) => getWordAdsErrorForSite( state, site ) );
+	const wordAdsSuccess = useSelector( ( state ) => getWordAdsSuccessForSite( state, site ) );
+	const isUnsafe = useSelector( ( state ) => isSiteWordadsUnsafe( state, siteId ) );
+	const canActivateWordAds = useSelector( ( state ) =>
+		canCurrentUser( state, siteId, 'activate_wordads' )
+	);
+	const requestingWordAdsApproval = useSelector( ( state ) =>
+		isRequestingWordAdsApprovalForSite( state, site )
+	);
+	const adsProgramName = useSelector( ( state ) =>
+		isJetpackSite( state, siteId ) ? 'Ads' : 'WordAds'
+	);
+
+	const canActivateWordadsInstant =
+		! site.options.wordads && canActivateWordAds && hasWordAdsFeature;
+	const canUpgradeToUseWordAds = ! site.options.wordads && ! hasWordAdsFeature;
+	const isWordadsInstantEligibleButNotOwner =
+		! site.options.wordads && hasWordAdsFeature && ! canActivateWordAds;
+	const isEnrolledWithIneligiblePlan =
+		site?.options?.wordads && ! hasWordAdsFeature && wordAdsStatus === WordAdsStatus.ineligible;
```

Comment:
For reviewers: This is a lot of lines, but all are just effectively moved from the connect method at the bottom of the file to here.